# HG changeset patch
# User Rik <rik@octave.org>
# Date 1390843116 28800
#      Mon Jan 27 09:18:36 2014 -0800
# Branch stable
# Node ID 1500ae1a1beafa3aa28d8ee5e6e6d9a4c6a71369
# Parent  e0b96a751afcf554a9bc105de7bebbc1f30692c2
__makeinfo__.m: Fix generating HTML output with Texinfo 5.2 (bug #41364).

* __makeinfo__.m: Add option "--output=-" so that file input will be sent to
stdout rather than re-written in-place which is the new default for 5.2.

diff --git a/scripts/help/__makeinfo__.m b/scripts/help/__makeinfo__.m
--- a/scripts/help/__makeinfo__.m
+++ b/scripts/help/__makeinfo__.m
@@ -124,20 +124,20 @@ function [retval, status] = __makeinfo__
       error ("__makeinfo__: could not create temporary file");
     endif
     fwrite (fid, text);
     fclose (fid);
 
     ## Take action depending on output type
     switch (lower (output_type))
       case "plain text"
-        cmd = sprintf ("%s --no-headers --no-warn --force --no-validate %s",
+        cmd = sprintf ("%s --no-headers --no-warn --force --no-validate --output=- %s",
                        makeinfo_program (), name);
       case "html"
-        cmd = sprintf ("%s --no-headers --html --no-warn --no-validate --force %s",
+        cmd = sprintf ("%s --no-headers --html --no-warn --no-validate --force --output=- %s",
                        makeinfo_program (), name);
       otherwise
         error ("__makeinfo__: unsupported output type: '%s'", output_type);
     endswitch
 
     ## Call makeinfo
     [status, retval] = system (cmd);
 
